lib/tasks/data/update_position_on_tools.rake

```diff
@@ -0,0 +1,19 @@
+namespace :data do
+  desc 'Update position on tools'
+  task :update_position_on_tools => :environment do |_, args|
+    puts 'Updating zines position'
+    Zine.all.update_all(position: 10)
+    Zine.where(buy_url: nil).or(Zine.where(buy_url: "")).update_all(position: 0)
```

Comment:
`@featured_tools` was all of the non-blank `buy_url`, right?

I think this will end up setting the previously featured tools to the bottom, but I think we want them at the top

I would just do 
```rb
Zine.where.not(buy_url: [nil, ""]).update_all(position: 1)
```

you could probably drop the `Zine.all.update_all(position: 10)` since if `position` is `nil` it will fall back to `published_at`

Reply:
This is the actual feature vs normal tools, I'm not sure I understand correctly what the code is doing since it's fetching the nil buy_url, but it's checking if the url is present. https://github.com/crimethinc/website/blob/main/app/controllers/tools_controller.rb#L44-L50

Reply:
the original code is saying 

```ruby
@featured_tools =  "all tools where buy_url is NOT nil or empty string"
@tools = "all tools where buy_url IS nil or empty string"
```
It doesn't really matter _why_ it was written with the odd usage of `.map` (maybe it was an effort to minimize DB calls?), what matters is that the end result is that tools with a buy_url show up at the top of the index page( e.g. `crimethinc.com/zines`).

the way this rake task is written, it will do the opposite. You set all zines to position 10, then you set zines **without** a `buy_url` to position 0.

0 comes before 10, we want the opposite. this only appears to work now because you changed the sortback to `desc` (i.e. 10 9 8 7 ..)

that is not how the position feature works on the rest of the site, it makes no sense for a user to put `100` in the admin UI, then have that show up at the top